server/leafnode.go

```diff
@@ -1053,7 +1056,7 @@ func (c *client) processLeafnodeInfo(info *Info) {
 				// allow access cross domain for each non-system account.
 				if hasJSDomain && acc.jetStreamConfigured() {
 					src := fmt.Sprintf(jsDomainAPI, opts.JetStreamDomain)
-					if err := acc.AddMapping(src, jsAllAPI); err != nil {
+					if err := acc.AddMapping(src, jsDenyAll); err != nil {
```

Comment:
Here, my expectation would be to see:
```
src := fmt.Sprintf("$JS.%s.>", opts.JetStreamDomain)
```
which would result in a mapping from `$JS.<domain>.>` to `$JS.>`, essentially stripping the domain.
Because we had more detailed denies earlier, messages sent with the prefix `$JS.<domain>` would not be denied.

Thus messages sent to anything with the prefix `$JS.<domain>.ACK` would not be blocked and see the domain stripped to be the prefix`$JS.ACK` instead.

For what is currently under API, it'd be `$JS.<domain>.API` -> `$JS.API`

Same for FC and SNAPSHOT